app/controllers/katello/concerns/api/v2/content_import_extensions.rb

```diff
@@ -0,0 +1,33 @@
+module Katello
+    module Concerns
+      module Api::V2::ContentImportExtensions
+        module ApiPieExtensions
+          extend ::Apipie::DSL::Concern
+  
+          update_api(:import_uploads) do
+            RepositoryTypeManager.generic_import_attributes.each do |import_attribute|
+              param import_attribute.api_param, import_attribute.type,
+                desc => N_(import_attribute.description)
+            end
+          end
+        end
+
+        extend ActiveSupport::Concern
+
+        included do
+          before_action :check_import_parameters, :only => [:import_uploads]
+        end
+
+        def check_import_parameters
+          repo_type = RepositoryTypeManager.find_by_content_type(params[:repository][:content_type])
+          repo_type.import_attributes.each do |import_attribute|
+            if import_attribute.required && params[import_attribute.api_param].blank?
+              fail HttpErrors::UnprocessableEntity, _("#{import_attribute.api_param} is required")
+            end
+          end
+        end
+          
+      end
+    end
+  end
```

Comment:
overall this looks good. I'm not sure i see a big benefit in making this a concern though. What was your line of thinking around that?